hashring/rbtree.go

```diff
@@ -85,25 +73,25 @@ func singleRotate(oldroot *RBNode, dir bool) *RBNode {
 	return newroot
 }
 
-func doubleRotate(root *RBNode, dir bool) *RBNode {
+func doubleRotate(root *Node, dir bool) *Node {
 	root.setChild(!dir, singleRotate(root.Child(!dir), !dir))
 	return singleRotate(root, dir)
 }
 
 // Insert inserts a value and string into the tree
 // Returns true on succesful insertion, false if duplicate exists
-func (t *RBTree) Insert(val int, str string) (ret bool) {
+func (t *RedBlackTree) Insert(val int, str string) (ret bool) {
```

Comment:
Likewise, should the RedBlackTree be public or an implementation detail of the "hash ring"?